src/Database/LSMTree/Internal/Index/Compact.hs

```diff
@@ -675,6 +660,27 @@ checkedBitVec off len ba
   | otherwise =
       Nothing
 
+{-------------------------------------------------------------------------------
+  Type class instantiation
+-------------------------------------------------------------------------------}
+
+instance Index IndexCompact where
+
+    search :: SerialisedKey -> IndexCompact -> PageSpan
+    search = search
+
+    sizeInPages :: IndexCompact -> NumPages
+    sizeInPages = sizeInPages
+
+    -- | The input bytestring must be 64 bit aligned. It is directly used as the
+    -- backing memory for the compact index.
```

Comment:
I’m not even sure what this constraint means. The first component of a serialized index is at index 0 and thus 64-bit-aligned, and where the other components are depends on the format of the index. Or does the above requirement refer to the memory address at which the content of the byte string starts? If yes, why is this 64 bit alignment required? Is it only for performance reasons? In this case, it wouldn’t be necessary to have it as a true requirement for all index types, documented with the type class, but could be mentioned as a recommendation here, in the module about *compact* indexes.

Reply:
I’ve dealt with this documentation as part of 559cc181bf7fd7114aba70d3c1604ab9000b8a52 now. I’ve settled on the interpretation that the whole contents of the byte string must be 64-bit-aligned and just specified this alignment as a general requirement, independent of the index type. Is this all okay?

Reply:
It means that the underlying byte array must be 64-bit aligned, which means the start address must be 64-bit aligned. It's important for performance (so we don't have to do unaligned memory operations). Maybe it's also important for correctness, I'm not sure if tests would fail currently if you use an unaligned bytestring.

In the current class setup, you wouldn't be able to figure out whether to pass an aligned or unaligned bytestring to the `fromSBS` function, without some additional information. It's probably simplest to require that a 64-bit aligned bytestring is passed to `fromSBS` regardless of the index type. The change you made in 559cc18 looks good to me.

Another approach is to add some extra info to the class, either through an additional class function or an associated type class, but I think the current approach of just requiring 64-bit alignment is sufficient.